static/app/views/monitors/components/overviewTimeline/gridLines.tsx

```diff
@@ -37,10 +38,11 @@ function getTimeMarkers(end: Date, timeWindow: TimeWindow, width: number): TimeM
   const times: TimeMarker[] = [];
   const start = getStartFromTimeWindow(end, timeWindow);
 
-  // Iterate and generate time markers which represent location of grid lines/time labels
+  const firstTimeMark = moment(start);
+  clampTimeBasedOnResolution(firstTimeMark, timeWindow);
```

Comment:
didn't change anything here, just tried to make the logic easier to follow by clamping the first reference tickmark, and then adding minutes starting from that